web/src/agents/node/get_last_sequence.rs

```diff
@@ -0,0 +1,41 @@
+use at2_node::client::{self, Client};
+use at2_ns::ThinUser;
+use wasm_bindgen_futures::spawn_local;
+use yew::worker::*;
+
+use crate::config::Config;
+
+/// Get the last sequence for a user
+pub struct GetLastSequence {
+    link: AgentLink<Self>,
+    client: Client,
+}
+
+impl Agent for GetLastSequence {
+    type Reach = Context<Self>;
+    type Message = (HandlerId, Result<sieve::Sequence, client::Error>);
+    type Input = ThinUser;
+    type Output = Result<sieve::Sequence, client::Error>;
+
+    fn create(link: AgentLink<Self>) -> Self {
+        let conf = Config::parse();
+
+        Self {
+            link,
+            client: Client::new(conf.network().to_owned()),
+        }
+    }
+
+    fn update(&mut self, (id, ret): Self::Message) {
+        self.link.respond(id, ret);
+    }
+
+    fn handle_input(&mut self, user: Self::Input, id: HandlerId) {
+        let mut client = self.client.clone();
+        let callback = self.link.callback(|ret| ret);
+
+        spawn_local(async move {
+            callback.emit((id, client.get_last_sequence(user.public_key()).await));
```

Comment:
Why does `get_last_sequence` modify the client?

Reply:
It uses a TCP stream internally and needs to modify it (incrementing pkg count, reading and writing to the stream, ...), so the client itself will need to mutate state to do so.

btw, as Rust disallow having double mutable reference, we need to `clone` the client for each new input, as the previous running `spawn_local` may not be finished at this point.